Develop

LukaszGryziewicz opened the pull request:
- [DEMO LINK](https://LukaszGryziewicz.github.io/layout_catalog-en/)
- [TEST REPORT LINK](https://LukaszGryziewicz.github.io/layout_catalog-en/report/html_report/)

LukaszGryziewicz committed:
added navigation

LukaszGryziewicz committed:
added cards, layout is complete, still need to fix some minor differences inside cards to pass tests

LukaszGryziewicz committed:
updated readme

LukaszGryziewicz committed:
applied what I learned in the lesson

LukaszGryziewicz committed:
addded width and border color as variables

LukaszGryziewicz committed:
made the same changes I did after review in previous task

meforoma commented on src/styles/blocks/content/card.scss:
```diff
@@ -0,0 +1,69 @@
+    display: flex;
+    justify-content: space-between;
+    padding-top: 16px;
+  }
```
agree with Swazi
consider adding
```
    line-height: 16px;
    align-items: center;
```

meforoma commented on src/styles/blocks/content/card-container.scss:
```diff
@@ -0,0 +1,17 @@
+  display: flex;
+  flex-wrap: wrap;
+  width: (4 * $card-width) + (3 * $card-gapX);
+  padding: 5px 0 70px;
```
it's better to set explicit top padding so that you don't need to combine it from multiple elments' styles
```suggestion
  padding: 50px 0 70px;
```
you'll then need to disable margin-top for the first 4 cards and you'll be good

LukaszGryziewicz replied:
We talked about all these line-heights Swazi mentioned in the review in DMs, my bad for not typing it out here, sorry.
Sadly they didn't help much.  Implemented the changes you recommended.

LukaszGryziewicz committed:
changes after review, sadly something is still off about the card

meforoma reviewed (changes requested):
@LukaszGryziewicz please double-check the imported font.
It seems to weight more than it should.
The import itself and then font-weights vs mock.

LukaszGryziewicz committed:
fixed font, added correct line heights, changed gapY and removed some redundant code

LukaszGryziewicz commented:
> @LukaszGryziewicz please double-check the imported font. It seems to weight more than it should. The import itself and then font-weights vs mock.

Hey, you were right about the fonts, the import was not the one I wanted. I added correct line heights everywhere (as in mock), but it still looked off. Then I changed gapY and it fixed the top row a lot, which is really weird to me.  Please take a look @meforoma

meforoma commented on src/styles/utils/variables.scss:
```diff
@@ -0,0 +1,15 @@
+$font-weight-bold: 700;
+$card-width: 200px;
+$card-gapX: 48px;
+$card-gapY: 51px;
```
this one is tricky - you need to have 50px from nav
and 46px between cards vertically

meforoma commented on src/styles/blocks/content/card.scss:
```diff
@@ -0,0 +1,69 @@
+
+  &__reviews {
+    padding-top: 16px;
+    line-height: 14px;
```
double-check this line-height i guess it should be 16px

meforoma reviewed (changes requested):
hope these two fixes should get you going

LukaszGryziewicz committed:
added Roman's suggestions, 4 additional tests pass

LukaszGryziewicz commented:
Changed gaps and line height, 4 additional tests pass. Don't really know what more can be wrong.

meforoma commented on src/styles/utils/variables.scss:
```diff
@@ -0,0 +1,15 @@
+$font-weight-medium: 500;
+$font-weight-bold: 700;
+$card-width: 200px;
+$card-gapX: 46px;
+$card-gapY: 50px;
```
try to use gaps only for spacing cards from each other, - use `Y: 48px`
and for spacing first cards row from nav menu - just apply padding-top on the cards container
![Snapshot 2022-03-31 23 08 19](https://user-images.githubusercontent.com/39308717/161141198-8b94c836-28ab-44f6-a656-3cc33b5972de.png)

LukaszGryziewicz committed:
removed gapY from first 4 cards, tweaked gap numbers

LukaszGryziewicz commented:
Ok, finally got it, thank you for all your suggestions @meforoma, turned out both X and Y gaps needed some tweaking.

meforoma reviewed (approved):
👍
